lightning/src/events/mod.rs

```diff
@@ -72,6 +72,16 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+impl PaymentPurpose {
+	/// Returns the preimage for this payment, if it is known.
+	pub fn preimage(&self) -> Option<PaymentPreimage> {
+		match self {
+			PaymentPurpose::InvoicePayment { payment_preimage, .. } => *payment_preimage,
```

Comment:
I know its in the original impl, but out of curiosity, 
in which scenarios we wouldnt have the payment_preimage in the traditional invoice?